src/main/kotlin/fi/metatavu/megasense/dataportal/airquality/PollutantController.kt

```
@@ -49,35 +53,59 @@ class PollutantController {
+                        id = UUID.randomUUID(),
+                        name = indexVariable.name,
+                        displayName = indexVariable.displayName,
+                        unit = foundUnit,
+                        creatorId = UUID.fromString(systemUserUUID),
+                        limit = null,
+                        lastModifierId = UUID.fromString(systemUserUUID),
+                        recommendation = null
+                    )
+                    logger.info("Imported new pollutant ${importedPollutant.displayName}")
+                } else if (foundPollutant.displayName != indexVariable.displayName) {
+                    val importedPollutant = pollutantDAO.updatePollutantDisplayName(
+                        pollutant = foundPollutant,
+                        displayName = indexVariable.displayName,
+                        modifierId = UUID.fromString(systemUserUUID)
+                    )
+                    logger.info("Update pollutant display name from ${foundPollutant.displayName} to ${importedPollutant.displayName}")
+                } else if (foundPollutant.unit != foundUnit) {
+                    val importedPollutant = pollutantDAO.updatePollutantUnit(
+                        pollutant = foundPollutant,
+                        unit = foundUnit,
+                        modifierId = UUID.fromString(systemUserUUID)
+                    )
+                    logger.info("Update pollutant unit from ${foundPollutant.unit} to ${importedPollutant.unit}")
+                }
             }
         }
     }
 
+    /**
+     * Gets unit of a pollutant from netcdf file
+     *
+     * @param netcdfFile opened netcdf file
+     * @param pollutantName pollutantName
+     * @return unit of a pollutant
+     */
+    private fun gePollutantUnits(netcdfFile: NetcdfFile, pollutantName: String): String {
+        return netcdfFile.findVariable(pollutantName)?.unitsString ?: "ug/m3"
```

if variable is not found, it is probably better to throw exception to show that the current configuration file is invalid for the data. 
If variable is not found -> exception
If variable if found, but there's no unit string -> default value